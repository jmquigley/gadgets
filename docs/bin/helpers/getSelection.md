<a name="getSelectionShim"></a>

## getSelectionShim()
DOM Traversal is not implemented in JSDOM
The best we can do is shim the functions

**Kind**: global function  
