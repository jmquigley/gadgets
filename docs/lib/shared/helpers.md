## Functions

<dl>
<dt><a href="#debug">debug(context, args)</a></dt>
<dd><p>A wrapper for the debug function.  It uses the package.json to enable or
disable debugging information in the library.  It also turns off
coverage checks for this function.</p>
</dd>
<dt><a href="#globalize">globalize(name, pkg, replace)</a> ⇒ <code>object</code></dt>
<dd><p>Takes a variable name and an object and places that name and associated
object into the global node space.  If the object is alread in the
global space, then that reference is used and returned.  The replace
flag is used to override the current global reference.</p>
</dd>
</dl>

<a name="debug"></a>

## debug(context, args)
A wrapper for the debug function.  It uses the package.json to enable or
disable debugging information in the library.  It also turns off
coverage checks for this function.

**Kind**: global function  

| Param | Type | Description |
| --- | --- | --- |
| context | <code>string</code> | the name of the module where this debug request is made. |
| args | <code>Array.&lt;object&gt;</code> | the list of arguments passed to the the debug function. |

<a name="globalize"></a>

## globalize(name, pkg, replace) ⇒ <code>object</code>
Takes a variable name and an object and places that name and associated
object into the global node space.  If the object is alread in the
global space, then that reference is used and returned.  The replace
flag is used to override the current global reference.

**Kind**: global function  
**Returns**: <code>object</code> - the global instance reference  

| Param | Type | Description |
| --- | --- | --- |
| name | <code>string</code> | the name of the variable to store globally |
| pkg | <code>object</code> | the object associated with the variable name |
| replace | <code>boolean</code> | if true, then it will always overwrite the current global (if it exists) |

