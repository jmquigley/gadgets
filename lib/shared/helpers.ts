"use strict";

const debug = require("debug")("helpers");

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is already in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(
	name: string,
	pkgname: any,
	replace: boolean = false
) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
		debug("Adding global variable %o -> %O", name, (global as any)[name]);
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
