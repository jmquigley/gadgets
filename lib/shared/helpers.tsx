'use strict';

const debugEntry = require('debug');
const pkg = require('../../package.json');

/**
 * A wrapper for the debug function.  It uses the package.json to enable or
 * disable debugging information in the library.  It also turns off
 * coverage checks for this function.
 * @param context {string} the name of the module where this debug request
 * is made.
 * @param args {object[]} the list of arguments passed to the the debug
 * function.
 */
export function debug(context: string, ...args: any[]) {
	/* istanbul ignore if  */
	if (pkg.debug) {
		const debugFn = debugEntry(`Gadgets -> ${context}`);
		debugFn(...args);
	}
}

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is alread in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(name: string, pkgname: any, replace: boolean = false) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
