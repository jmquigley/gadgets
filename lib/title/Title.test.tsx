'use strict';

import {cleanup, log, mockupEnv} from '../../test/helpers';
mockupEnv();

import test from 'ava';
import {mount} from 'enzyme';
import * as path from 'path';
import * as React from 'react';
import {getDefaultTitleProps, Title} from './index';

test.after.always.cb(t => {
	cleanup(path.basename(__filename), t);
});

test('Test retrieval of Title props object', t => {
	const props = getDefaultTitleProps();

	t.true('stacked' in props);
	t.false(props.stacked);

	t.true('widget' in props);
	t.is(props.widget, null);
});

test('Test creation of a Title control', t => {
	let s: string = 'Test label text';
	const ctl = mount(<Title className="test-class">{s}</Title>);

	t.truthy(ctl);
	log.debug(ctl.html(), __filename);

	t.is(ctl.prop('id'), '');
	t.false(ctl.prop('disabled'));
	t.true(ctl.prop('visible'));
	t.is(ctl.find('.ui-title').length, 1);
	t.is(ctl.find('.test-class').length, 1);
	t.is(ctl.text(), s);
});
