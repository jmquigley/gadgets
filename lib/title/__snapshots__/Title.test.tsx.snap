// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a Title control with bad layout 1`] = `
<Title
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  layout={9999}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={true}
  notheme={false}
  notooltip={false}
  obj="Title"
  onClick={[Function]}
  onUpdate={[Function]}
  padding={null}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  title="Test label text"
  tooltip={null}
  top={null}
  useedit={false}
  visible={true}
  widget={null}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a Title control with bad layout 2`] = `
<Title
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  layout={9999}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={true}
  notheme={false}
  notooltip={false}
  obj="Title"
  onClick={[Function]}
  onUpdate={[Function]}
  padding={null}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  title="Test label text"
  tooltip={null}
  top={null}
  useedit={false}
  visible={true}
  widget={null}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a Title control with default props 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  layout={5}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={true}
  notheme={false}
  notooltip={false}
  obj="Title"
  onClick={[Function]}
  onError={[Function]}
  onUpdate={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  title="Test label text"
  tooltip={null}
  top={null}
  useedit={false}
  visible={true}
  widget={null}
  width={null}
  xcss={null}
>
  <Title__TitleView
    className="ui-title-bar test-class"
    disabled={false}
    layout={5}
    sizing="normal"
    style={Object {}}
    visible={true}
  >
    <Title__StyledLabel
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={5}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      onUpdate={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      text="Test label text"
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget={null}
      width={null}
      xcss={
        Array [
          "text-align:left;flex:5;",
        ]
      }
    />
  </Title__TitleView>
</Wrapper>
`;

exports[`Test creation of a Title control with even layout 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  layout={1}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={true}
  notheme={false}
  notooltip={false}
  obj="Title"
  onClick={[Function]}
  onError={[Function]}
  onUpdate={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  title="Test label text"
  tooltip={null}
  top={null}
  useedit={false}
  visible={true}
  widget="widget"
  width={null}
  xcss={null}
>
  <Title__TitleView
    className="ui-title-bar"
    disabled={false}
    layout={1}
    sizing="normal"
    style={Object {}}
    visible={true}
  >
    <Title__StyledLabel
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={1}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      onUpdate={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      text="Test label text"
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget="widget"
      width={null}
      xcss={
        Array [
          "text-align:left;flex:1;",
        ]
      }
    />
    <Title__StyledWidget
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title-widget"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={1}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget="widget"
      width={null}
      xcss={
        Array [
          "text-align:right;flex:1;",
        ]
      }
    >
      widget
    </Title__StyledWidget>
  </Title__TitleView>
</Wrapper>
`;

exports[`Test creation of a Title control with no layout 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  layout={6}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={true}
  notheme={false}
  notooltip={false}
  obj="Title"
  onClick={[Function]}
  onError={[Function]}
  onUpdate={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  title="Test label text"
  tooltip={null}
  top={null}
  useedit={false}
  visible={true}
  widget={null}
  width={null}
  xcss={null}
>
  <Title__TitleView
    className="ui-title-bar"
    disabled={false}
    layout={6}
    sizing="normal"
    style={Object {}}
    visible={true}
  >
    <Title__StyledLabel
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={6}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      onUpdate={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      text="Test label text"
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget={null}
      width={null}
      xcss={
        Array [
          "text-align:left;flex:1;",
        ]
      }
    />
  </Title__TitleView>
</Wrapper>
`;

exports[`Test creation of a Title control with quarter layout 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  layout={0}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={true}
  notheme={false}
  notooltip={false}
  obj="Title"
  onClick={[Function]}
  onError={[Function]}
  onUpdate={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  title="Test label text"
  tooltip={null}
  top={null}
  useedit={false}
  visible={true}
  widget="widget"
  width={null}
  xcss={null}
>
  <Title__TitleView
    className="ui-title-bar"
    disabled={false}
    layout={0}
    sizing="normal"
    style={Object {}}
    visible={true}
  >
    <Title__StyledLabel
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={0}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      onUpdate={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      text="Test label text"
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget="widget"
      width={null}
      xcss={
        Array [
          "text-align:left;flex:1;",
        ]
      }
    />
    <Title__StyledWidget
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title-widget"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={0}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget="widget"
      width={null}
      xcss={
        Array [
          "text-align:right;flex:3;",
        ]
      }
    >
      widget
    </Title__StyledWidget>
  </Title__TitleView>
</Wrapper>
`;

exports[`Test creation of a Title control with stacked layout 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  layout={4}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={true}
  notheme={false}
  notooltip={false}
  obj="Title"
  onClick={[Function]}
  onError={[Function]}
  onUpdate={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  title="Test label text"
  tooltip={null}
  top={null}
  useedit={false}
  visible={true}
  widget="widget"
  width={null}
  xcss={null}
>
  <Title__TitleView
    className="ui-title-bar"
    disabled={false}
    layout={4}
    sizing="normal"
    style={Object {}}
    visible={true}
  >
    <Title__StyledLabel
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={4}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      onUpdate={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      text="Test label text"
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget="widget"
      width={null}
      xcss={
        Array [
          "display:block;",
        ]
      }
    />
    <Title__StyledWidget
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title-widget"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={4}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget="widget"
      width={null}
      xcss={
        Array [
          "display:block;",
        ]
      }
    >
      widget
    </Title__StyledWidget>
  </Title__TitleView>
</Wrapper>
`;

exports[`Test creation of a Title control with thirds layout 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  layout={3}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={true}
  notheme={false}
  notooltip={false}
  obj="Title"
  onClick={[Function]}
  onError={[Function]}
  onUpdate={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  title="Test label text"
  tooltip={null}
  top={null}
  useedit={false}
  visible={true}
  widget="widget"
  width={null}
  xcss={null}
>
  <Title__TitleView
    className="ui-title-bar"
    disabled={false}
    layout={3}
    sizing="normal"
    style={Object {}}
    visible={true}
  >
    <Title__StyledLabel
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={3}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      onUpdate={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      text="Test label text"
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget="widget"
      width={null}
      xcss={
        Array [
          "text-align:left;flex:1;",
        ]
      }
    />
    <Title__StyledWidget
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title-widget"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={3}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget="widget"
      width={null}
      xcss={
        Array [
          "text-align:right;flex:2;",
        ]
      }
    >
      widget
    </Title__StyledWidget>
  </Title__TitleView>
</Wrapper>
`;

exports[`Test creation of a Title control with threequarter layout 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  layout={2}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={true}
  notheme={false}
  notooltip={false}
  obj="Title"
  onClick={[Function]}
  onError={[Function]}
  onUpdate={[Function]}
  padding={null}
  reset={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  title="Test label text"
  tooltip={null}
  top={null}
  useedit={false}
  visible={true}
  widget="widget"
  width={null}
  xcss={null}
>
  <Title__TitleView
    className="ui-title-bar"
    disabled={false}
    layout={2}
    sizing="normal"
    style={Object {}}
    visible={true}
  >
    <Title__StyledLabel
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={2}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      onUpdate={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      text="Test label text"
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget="widget"
      width={null}
      xcss={
        Array [
          "text-align:left;flex:3;",
        ]
      }
    />
    <Title__StyledWidget
      backgroundColor={null}
      borderColor={null}
      bottom={null}
      className="ui-title-widget"
      color={null}
      contentEditable={false}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      id={null}
      layout={2}
      left={null}
      location="none"
      minHeight={null}
      minWidth={null}
      noedit={false}
      nohover={false}
      noripple={true}
      notheme={false}
      notooltip={false}
      obj="Title"
      onClick={[Function]}
      padding={null}
      selected={false}
      sizing="normal"
      style={Object {}}
      testing={true}
      theme={null}
      title="Test label text"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget="widget"
      width={null}
      xcss={
        Array [
          "text-align:right;flex:1;",
        ]
      }
    >
      widget
    </Title__StyledWidget>
  </Title__TitleView>
</Wrapper>
`;

exports[`Test disabling the Title control 1`] = `
<Title
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={true}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  layout={5}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={true}
  notheme={false}
  notooltip={false}
  obj="Title"
  onClick={[Function]}
  onUpdate={[Function]}
  padding={null}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  title="Test label text"
  tooltip={null}
  top={null}
  useedit={false}
  visible={true}
  widget={null}
  width={null}
  xcss={null}
/>
`;

exports[`Test making the Title control invisible 1`] = `
<Title
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  layout={5}
  left={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  noripple={true}
  notheme={false}
  notooltip={false}
  obj="Title"
  onClick={[Function]}
  onUpdate={[Function]}
  padding={null}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={null}
  title="Test label text"
  tooltip={null}
  top={null}
  useedit={false}
  visible={false}
  widget={null}
  width={null}
  xcss={null}
/>
`;

exports[`Test retrieval of Title props object 1`] = `
Object {
  "backgroundColor": null,
  "borderColor": null,
  "bottom": null,
  "children": null,
  "className": null,
  "color": null,
  "contentEditable": false,
  "controlled": true,
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "id": null,
  "layout": 5,
  "left": null,
  "location": "none",
  "minHeight": null,
  "minWidth": null,
  "noedit": false,
  "nohover": false,
  "noripple": true,
  "notheme": false,
  "notooltip": false,
  "obj": "Title",
  "onClick": [Function],
  "onUpdate": [Function],
  "padding": null,
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": null,
  "tooltip": null,
  "top": null,
  "useedit": false,
  "visible": true,
  "widget": null,
  "width": null,
  "xcss": null,
}
`;

exports[`Test retrieval of Title props object 2`] = `
Object {
  "children": null,
  "classes": b {
    "_classes": Map {
      "ui-title-bar" => true,
    },
    "_clstr": "",
    "_delimiter": " ",
    "_dirty": true,
    "_obj": null,
    "delete": [Function],
    "reset": [Function],
  },
  "sizing": "normal",
  "style": Object {},
}
`;
